internal/sidecarlogresults/sidecarlogresults.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sidecarlogresults
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"golang.org/x/sync/errgroup"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// ErrorReasonMaxResultSizeExceeded indicates that the result exceeded its maximum allowed size
+// var ErrorReasonMaxResultSizeExceeded = fmt.Errorf("%s", "MaxResultSizeExceeded")
+var ErrorReasonMaxResultSizeExceeded = "MaxResultSizeExceeded"
+
+// SidecarLogResult holds fields for storing extracted results
+type SidecarLogResult struct {
+	Name  string
+	Value string
+}
+
+func fileExists(filename string) (bool, error) {
+	info, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false, nil
+	} else if err != nil {
+		return false, err
+	}
+	return !info.IsDir(), nil
+}
+
+func waitForStepsToFinish(runDir string) error {
+	steps := make(map[string]bool)
+	files, err := ioutil.ReadDir(runDir)
+	if err != nil {
+		return err
+	}
+	for _, file := range files {
+		steps[filepath.Join(runDir, file.Name(), "out")] = true
+	}
+	for true {
+		for stepFile := range steps {
+			// check if there is a post file without error
+			exists, err := fileExists(stepFile)
+			if err != nil {
+				return err
+			}
+			if exists {
+				delete(steps, stepFile)
+				continue
+			}
+			// check if there is a post file with error
+			exists, err = fileExists(fmt.Sprintf("%s.err", stepFile))
+			if err != nil {
+				return err
+			}
+			// there is an error file meaning that the steps later will not run.
+			if exists {
+				return nil
+			}
+		}
+		if len(steps) == 0 {
```

Comment:
Instead of putting this here, can you make line 68:

`for len(steps) != 0` ?

Reply:
I don't think I can because in line 76, I'm deleting a step after it has completed. This condition is so that we can break out of the infinite loop as soon as all the steps have finished running.